### PR TITLE
Add IdT, the identity monad transformer

### PR DESCRIPTION
**Pull request by vikraman**

### CODE REVIEW

**Comment by vikraman** [comment]
I don't see a `MonadTrans` yet, should that be added?

**Comment by ceedubs** [comment]
Thanks, @vikraman!

I've never used `IdT` before. Do you have an example of how/why one might want to use it?

**Comment by xuwei-k** [comment]
I think `IdT` should be an own data type, not type alias.

**Comment by vikraman** [comment]
Yes, I just ran into diverging implicits with this alias, I'll change it to a proper data type.

**Comment by vikraman** [comment]
Updated.

**Comment by refried** [comment]
Ditto to @ceedubs's question

File: core/src/main/scala/cats/data/IdT.scala [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+/**
+ * `IdT[F[_], A]` is the identity monad transformer.
+ */
+final case class IdT[F[_], A](value: F[A]) {
+
+  def map[B](f: A => B)(implicit F: Functor[F]): IdT[F, B] =
+    IdT(F.map(value)(f))
+
+  def flatMap[B](f: A => IdT[F, B])(implicit F: Monad[F]): IdT[F, B] =
```
Comment (julien-truffaut) [inline core/src/main/scala/cats/data/IdT.scala]: should be replace the constraint from `Monad` to `FlatMap`?

**Comment by vikraman** [comment]
I think it is useful to have `IdT` since it is the most trivial transformer. It would be useful to have the definitions of `liftM` once there is a `MonadTrans` class.

**Comment by xuwei-k** [comment]
https://github.com/xuwei-k/cats/commit/5b426cc249bd1f46d92b8d2

```
[error] /home/travis/build/xuwei-k/cats/tests/src/test/scala/cats/tests/IdTTests.scala:10: ambiguous implicit values:
[error]  both method idTFunctor in class IdTInstances of type [F[_]](implicit F: cats.Functor[F])cats.Functor[[β]cats.data.IdT[F,β]]
[error]  and method idTMonad in class IdTInstances of type [F[_]](implicit F: cats.Monad[F])cats.Monad[[β]cats.data.IdT[F,β]]
[error]  match expected type cats.Functor[[β]cats.data.IdT[F,β]]
[error]   def idTFunctor[F[_]: Monad] = implicitly[Functor[IdT[F, ?]]]
[error]                                           ^
```

**Comment by vikraman** [comment]
Split the instances to avoid ambiguous implicits.

**Comment by non** [comment]
Sorry for the long silence.

I'm fine with merging this -- I do think we probably want to write a few more words about how it works (e.g. "If M[_] is a monad, then IdT[M, ?] provides an equivalent monad" or similar).

:+1:

**Comment by julien-truffaut** [comment]
LGTM 👍

**Pull request merged**